onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -881,7 +887,27 @@ class PlannerImpl {
           if (!node_output->Exists()) continue;
           OrtValueIndex index = Index(node_output->Name());
           ProcessDef(index, node_output);
-          plan_.SetLocation(static_cast<size_t>(index), exec_provider->GetOrtDeviceByMemType(p_kernel_def->OutputMemoryType(i)));
+          OrtDevice output_device = exec_provider->GetOrtDeviceByMemType(p_kernel_def->OutputMemoryType(i));
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+          // Downstream nodes of certain providers may require a CPU accessible location override
+          // to make sure the EP does not incur an unnecessary copy.
```

Comment:
Naively 'accessible' would be shared GPU memory (accessible from CPU) vs. dedicated (not accessible). 

Is this about it being a CPU 'accessible location' or that the memory has a specific alignment requirement?

Reply:
This could be both or one of them. The point here, is that certain EPs would like to prevent copies and for that they want their input allocated using the allocator of their choice. It could be an aligned memory, or it could something like GPU Pinned. This also depends on the EP configuration.